service/history/queues/scheduler.go

```diff
@@ -257,3 +257,13 @@ func (s *rateLimitedSchedulerImpl) Stop() {
 func (s *rateLimitedSchedulerImpl) TaskChannelKeyFn() TaskChannelKeyFn {
 	return s.baseScheduler.TaskChannelKeyFn()
 }
+
+// Less implements comparison for TaskChannelKey for btree
+// Priority is the primary ordering, NamespaceID is the tie-breaker to ensure unique keys
+func (t TaskChannelKey) Less(other TaskChannelKey) bool {
+	if t.Priority != other.Priority {
+		return t.Priority < other.Priority
+	}
+	// Use NamespaceID as tie-breaker to ensure all keys are distinguishable
+	return t.NamespaceID < other.NamespaceID
```

Comment:
Let's probably sync offline on this. I still think this will cause starvation for customers with "bigger" namespaceID when task scheduler encounters host/cell level limit. 

Task scheduler rate limiter are all namespace level today so we are fine from that perspective. 
But the task processing worker pool size may become the issue especially for Visibility queue where each task can take a long time when ES is slow. Then only tasks with small Namespace can be processed.